src/Lib.P2P/Cid.cs

```diff
@@ -0,0 +1,584 @@
+
+                case "L":
+                    var sb = new StringBuilder();
+                    sb.Append(Encoding);
+                    sb.Append(' ');
+                    sb.Append("cidv");
+                    sb.Append(Version);
+                    sb.Append(' ');
+                    sb.Append(ContentType);
+                    if (Hash != null)
+                    {
+                        sb.Append(' ');
+                        sb.Append(Hash.Algorithm.Name);
+                        sb.Append(' ');
+                        sb.Append(MultiBase.Encode(Hash.ToArray(), Encoding).Substring(1));
+                    }
+
+                    return sb.ToString();
+
+                default:
+                    throw new FormatException($"Invalid CID format specifier '{format}'.");
+            }
+        }
+
+        /// <summary>
+        ///   Converts the CID to its equivalent string representation.
+        /// </summary>
+        /// <returns>
+        ///   The string representation of the <see cref="Cid"/>.
+        /// </returns>
+        /// <remarks>
+        ///   For <see cref="Version"/> 0, this is equalivalent to the 
+        ///   <see cref="MultiHash.ToBase58()">base58btc encoding</see>
+        ///   of the <see cref="Hash"/>.
+        /// </remarks>
+        /// <seealso cref="Decode"/>
+        public string Encode()
+        {
+            if (encodedValue != null) return encodedValue;
+            if (Version == 0)
+                encodedValue = Hash.ToBase58();
+            else
+                using (var ms = new MemoryStream())
+                {
+                    ms.WriteVarint(Version);
+                    ms.WriteMultiCodec(ContentType);
+                    Hash.Write(ms);
+                    encodedValue = MultiBase.Encode(ms.ToArray(), Encoding);
+                }
+
+            return encodedValue;
+        }
+
+        /// <summary>
+        ///   Converts the specified <see cref="string"/>
+        ///   to an equivalent <see cref="Cid"/> object.
+        /// </summary>
+        /// <param name="input">
+        ///   The <see cref="Cid.Encode">CID encoded</see> string.
+        /// </param>
+        /// <returns>
+        ///   A new <see cref="Cid"/> that is equivalent to <paramref name="input"/>.
+        /// </returns>
+        /// <exception cref="FormatException">
+        ///   When the <paramref name="input"/> can not be decoded.
+        /// </exception>
+        /// <seealso cref="Encode"/>
+        public static Cid Decode(string input)
+        {
+            try
+            {
+                // SHA2-256 MultiHash is CID v0.
+                if (input.Length == 46 && input.StartsWith("Qm")) return (Cid) new MultiHash(input);
+
+                using (var ms = new MemoryStream(MultiBase.Decode(input), false))
+                {
+                    var v = ms.ReadVarint32();
+                    if (v != 1) throw new InvalidDataException($"Unknown CID version '{v}'.");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)